### ISSUE TITLE
RSpec 3.2 optimisation

### ISSUE THREAD

**Issue by florrain** [rspec/rspec-core#1956]
Hello,

Just switched from RSpec 3.0 to 3.2 and I'm astonished by the huge optimisation. The suite running time of my project went from 10min to 6min! This is local.

One thing is that I didn't notice the optimisation on a remote Jenkins. The running time is exactly the same. I was wondering if it because of the hardware specs of the machine? (single-cpu, low memory, ...) and what do you recommend to improve the performance in terms of environment setup?

Thanks

**Comment by JonRowe** [comment]
It's hard to say what could be causing the lack of additional performance on your CI box, it could be many things. One thing I've noticed in the past is that cheaper CPUs can lack features making Ruby in general slower (lack of out of order execution in particular being a real bottleneck) so it could be your CPU isn't beefy enough (cores won't matter because Ruby is in general single threaded) or you could be maxing out your Ram on CI still but your local machine now has breathing room with reduced allocations. Another factor could be that your CI environment uses a convential hard drive where as many laptops now use SSD.

In short, there are two many variables here to comment :)
